product.go

```diff
@@ -72,12 +78,99 @@ type ProductsResource struct {
 	Products []Product `json:"products"`
 }
 
+// Pagination of results
+type Pagination struct {
+	NextPageOptions		*ListOptions
+	PreviousPageOptions *ListOptions
+}
+
 // List products
 func (s *ProductServiceOp) List(options interface{}) ([]Product, error) {
+	products, _, err := s.ListWithPagination(options)
+	return products, err
+}
+
+// List products and return pagination to retrieve next/previous results
+func (s *ProductServiceOp) ListWithPagination(options interface{}) ([]Product, Pagination, error) {
 	path := fmt.Sprintf("%s/%s.json", globalApiPathPrefix, productsBasePath)
 	resource := new(ProductsResource)
-	err := s.client.Get(path, resource, options)
-	return resource.Products, err
+	headers := http.Header{}
+
+	err := s.client.CreateAndDo("GET", path, nil, options, resource, &headers)
+
+	// Extract pagination from header
+	linkHeader := ""
+	_, ok := headers["Link"]
+	if ok {
+		linkHeader = headers["Link"][0]
+	}
+	pagination, err := extractPagination(linkHeader)
+
```

Comment:
Generally you check the error and if it is not nil return explicit nils or empty values for other return values.

ie replace this return line with

```
if err != nil {
return nil, Paginaton{}, err
}
return resource.Products, pagination, nil
```